### PR TITLE
SAA-514 pageable endpoint to retrieve events recorded as a result of changing circumstances

### PR DESCRIPTION
**Pull request by timharrison-moj**
Will return a paginated list of events for a prison relating to:

* Cell moves
* Non-association details amended
* Incentive level changes

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/EventReviewServiceTest.kt [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    val searchSpec = EventReviewSearchRequest(prisonCode = prisonCode, eventDate = date, prisonerNumber = prisonerNumber, acknowledgedEvents = true)
+
+    val result = eventReviewService.getFilteredEvents(page, size, sortDirection, searchSpec)
+
+    assertThat(result.totalElements).isEqualTo(3)
+    assertThat(result.content).hasSize(3)
+    assertThat(result.totalPages).isEqualTo(1)
+
+    result.content.map {
+      assertThat(it.eventReviewId in 1..3)
```
Comment (matt-crowson-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/EventReviewServiceTest.kt:55]: Is this assertion on the range `1..3` missing an `.isTrue` ?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/integration/EventReviewIntegrationTest.kt [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+  fun `should paginate events`() {
+    var result = webTestClient.getEvents(page = 0, size = 2)
+      .expectStatus().isOk
+      .expectHeader().contentType(MediaType.APPLICATION_JSON)
+      .expectBody(EventReviewSearchResults::class.java)
+      .returnResult().responseBody
+
+    assertThat(result).isNotNull
+
+    with(result!!) {
```
Comment (matt-crowson-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/integration/EventReviewIntegrationTest.kt:51]: I would just move the !! to the end of line 47 and do away with the non null check. Meh.

`.returnResult().responseBody!!`

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/resource/EventReviewController.kt [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class),
+          ),
+        ],
+      ),
+    ],
+  )
+  fun getEventsForReview(
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/resource/EventReviewController.kt:62]: Could add some validation annotations to the request params, meh?

```
prisonCode -> NotBlank
date -> PastOrPresent
page, size -> PositiveOrZero
```

**Review by matt-crowson-moj — APPROVED** [review]
LGTM

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/EventReviewServiceTest.kt (hunk 1 above)
Comment (timharrison-moj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/EventReviewServiceTest.kt:55]: Yes. Added now.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/resource/EventReviewController.kt (hunk 3 above)
Comment (timharrison-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/resource/EventReviewController.kt:62]: Yes agreed. Added validation checks to the mandatory request values

**Review by matt-crowson-moj — APPROVED** [review]
LGTM

**Pull request merged**